lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -78,11 +92,13 @@ void ExecutionEngine::run(Context &ctx) {
   assert(function_ && "No function has been compiled");
   // Make sure that the context has backing tensors for all placeholders.
   ctx.allocate(M_.getPlaceholders());
-  function_->setupRuns();
+  if (firstRun_) {
+    function_->setupRuns();
```

Comment:
The introduction of `firstRun_` is to allow us to do later setup of the function? What is the benefit of this?